### ISSUE TITLE
MPV not working as expected

### ISSUE THREAD

**Issue by ciaens** [jarnedemeulemeester/findroid#645]
**Describe the bug**
Depending on the configuration, the MPV backend doesn't work, or exhibit color shift, color banding, compression, and fails to display an HDR image.

**To Reproduce**
1. Go to a movie encoded in HEVC 10 bit with HDR10 (main).
2. Click on play
4. See error (black screen or wrong playback).

**Expected behavior**
Normal playback of the movie, as done with the default backend (Exoplayer).

**Screenshots**
![Screenshot_20240131-201215](https://github.com/jarnedemeulemeester/findroid/assets/111684368/b9c5c761-a97f-4577-93c6-30a9e69b3e8a)
![Screenshot_20240131-201246](https://github.com/jarnedemeulemeester/findroid/assets/111684368/98ed259e-d2fc-4dcc-ba72-938fd4199795)

First image is MPV, second is ExoPlayer. You may need to increase brightness as it was captured on an HDR screen. Very noticeable in person.

**Device info (please complete the following information):**
 - Device: Google Pixel 8 Pro
 - Android version: 14 / UQ1A.240105
 - App version: v0.13.1 / versionCode 22 / targetSDK 34
 - Jellyfin version: not relevant

**Additional context**
Here are some logs where I start by using MPV, and then switch to ExoPlayer.
In this first log,  I'm using first MPV with gpu-next and mediacodec-copy, then ExoPlayer
`E dev.jdtech.jellyfin: PIXEL: ioctl err: 1` is spammed during MPV playback. See complete verbose logs below:
[Findroid log 8242f4cf56ba.txt](https://github.com/jarnedemeulemeester/findroid/files/14124563/Findroid.log.8242f4cf56ba.txt)

In this second log, I'm first using MPV with no mediacopy / hardware encoding, then switching to mediacodec-copy.
Related issue: https://github.com/jarnedemeulemeester/findroid/issues/537
When disabling hardware acceleration, I only get a black screen. When using the hardware encoding mediacodec or mediacodec-copy, I have an image (it's not displaying correctly thus why I'm making this issue).
[Findroid log 6c1e37110e7e.txt](https://github.com/jarnedemeulemeester/findroid/files/14124570/Findroid.log.6c1e37110e7e.txt)

About this: https://github.com/jarnedemeulemeester/findroid/issues/574
I've sometime encountered this issue, but this was either weird encoded movies, or solved by clearing the cache of the application.

**Comment by jarnedemeulemeester** [comment]
The differences between Findroid and mpv-android may be because mpv-android is using a newer version of MPV than Findroid.

The next version of Findroid will upgrade mpv to the latest version (0.37.0) which will hopefully solve the tonemapping issues you're seeing.

And yeah, it's correct that mpv in Findroid has the same limitations (HDR) as mpv-android.

**Comment by Jcuhfehl** [comment]
@ciaens What video settings do you use in mpv android? Findroid uses mpv's `fast` profile which leads to lower quality. Can you test if changing your mpv android settings to the following recreates the artifacts you're experiencing:
```
scale=bilinear
dscale=bilinear
dither=no
correct-downscaling=no
linear-downscaling=no
sigmoid-upscaling=no
hdr-compute-peak=no
allow-delayed-peak-detect=yes
```
I plan to create a pr to allow switching between mpv profiles (probably also individual settings) because there are a lot of quality improvements to be had from using actually good scaling kernels

**Comment by Jcuhfehl** [comment]
This honestly just looks like different tonemapping methods. Personally vlc (are you sure it's vlc and not exoplayer?) seems worst (it nukes all the stars but I'm not sure because they're not the same frame). Providing more options is basically the only solution, since there isn't a "mathematically correct" way to tonemap so it's always going to come down to subjective opinion

**Comment by Jcuhfehl** [comment]
@ciaens could you test with #652 (with high quality enabled) and gpu-next to see if you still get the same issue?

**Comment by GrabbenD** [comment]
> Mpv on android cannot support hdr. This means all playback is tonemapped

Is this just Android or also Android TV?

**Comment by Jcuhfehl** [comment]
It's an mpv limitation, so it applies to both android and android tv

**Comment by Doc-4** [comment]
When you ship the next version of Findroid, can you ship it with [this sub branch](https://github.com/mpv-android/mpv-android/tree/sfan5/vk) of MPV if possible?

It supports HDR for compatible devices using the flags mentioned in this issue https://github.com/mpv-android/mpv-android/issues/874#issuecomment-2242079740

**Comment by foux** [comment]
It would be great to have the HDR compatible MPV version

**Comment by Doc-4** [comment]
Having used it plenty since back then, I'd actually advise shipping it as a separate version or test at first instead, as MPV android vulkan version can't use HW+ (mediacodec-copy) and as a result tends to stutter on large files with high bitrates using HW (this just so happens to be a lot of HDR content), even on powerful devices.

**Comment by foux** [comment]
You use it with FIndroid? SO I guess you made your own build?

**Comment by Doc-4** [comment]
> You use it with FIndroid? SO I guess you made your own build?

No, but you can obviously use other Jellyfin clients that support using external players to just launch that version of MPV configured for HDR.
